sdk/keyvault/keyvault-keys/migration-guide.md

```diff
@@ -0,0 +1,251 @@
+```
+
+Below is a simple example using both `@azure/keyvault-keys` and [`DefaultAzureCredential`][identity-readme-DAC]:
+
+```ts
+// The default credential first checks environment variables for configuration as described above.
+// If environment configuration is incomplete, it will try managed identity.
+const { KeyClient } = require("@azure/keyvault-keys");
+const { DefaultAzureCredential } = require("@azure/identity");
+
+// Azure SDK clients accept the credential as a parameter
+const credential = new DefaultAzureCredential();
+const vaultUrl = "https://my-vault.vault.azure.net/";
+const client = new KeyClient(vaultUrl, credential);
+
+async function main(): Promise<void> {
+  const keyVaultKey = await client.getKey("MyKey");
+  console.log(keyVaultKey);
+}
+
+main().catch((err) => console.error(err));
+```
+
+The new client also provides a `CryptographyClient` to perform cryptographic operations (encrypt/decrypt, wrap/unwrap, sign/verify) using a particular key. Below is an example.
+
+```ts
+const { KeyClient, CryptographyClient } = require("@azure/keyvault-keys");
+const { DefaultAzureCredential } = require("@azure/identity");
+
+const credential = new DefaultAzureCredential();
+const vaultUrl = "https://my-vault.vault.azure.net/";
+const client = new KeyClient(vaultUrl, credential);
+
+async function main(): Promise<void> {
+  const keyVaultKey = await client.getKey("MyKey");
+  const cryptographyClient = new CryptographyClient(keyVaultKey.id!, credential);
+  const signed = await cryptographyClient.signData("RS256", "some data");
+  console.log(signed);
+}
+
+main().catch((err) => console.error(err));
+```
+
+### Create a key
+
+In `azure-keyvault` you could create a key by using `KeyVaultClient`'s `createKey` method, which required a vault endpoint, key name, and key type. This method returned a `KeyBundle` containing the key.
+
+```js
+// create an RSA key
+let keyBundle = await client.createKey(vaultUrl, "myRSAKey", "RSA");
+console.log(keyBundle);
+
+// create an elliptic curve key
+keyBundle = await client.createKey(vaultUrl, "myECKey", "EC")
+console.log(keyBundle);
+```
+
+Now in `@azure/keyvault-keys` there are multiple ways to create keys. You can provide a key name and type to the general `createKey` method, or provide just a name to `createRsaKey` or `createEcKey`. These methods all return the created key as a `KeyVaultKey`.
+
+```ts
+const rsaKey1 = await client.createKey("MyRSAKey1", "RSA");
+const rsaKey2 = await client.createRsaKey("MyRSAKey2");
+const ecKey1 = await client.createKey("MyECKey1", "EC");
+const ecKey2 = await client.createEcKey("MyECKey2");
+```
+
+### Retrieve a key
+
+In `azure-keyvault` you could retrieve a key (in a `KeyBundle`) by using `getKey` and specifying the desired vault endpoint, key name, and key version. You could retrieve the versions of a key with the `getKeyVersions` method, which returned an array with key items that can be used to retrieve the full key afterwards.
```

Comment:
The selling point here for me is that in the old package, you had to know the key version, otherwise there was no way to get the key value. So, I would put emphasis there.


```suggestion
In `azure-keyvault` you could retrieve a key only if you knew both the key name and version. If you did not know the version, you would first use the `getKeyVersions` method to get all versions of the key and then pass the name and version to the `getKey` method. Both methods would need the vault endpoint as well.
```

Reply:
This is not true since an empty string also works to get the latest version. 🤔